test_collections/sdk_tests/support/yaml_tests/models/yaml_test_models.py

```diff
@@ -13,38 +13,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-from enum import Enum
-from pathlib import Path
-from typing import Any, Optional
+from typing import Any
 
-from pydantic import BaseModel, Field
 from pydantic_yaml import YamlModelMixin
 
+from ...models.matter_test_models import MatterTest, MatterTestType
+
 ###
 # This file declares YAML models that are used to parse the YAML Test Cases.
 ###
 
 
-class YamlTestType(Enum):
-    AUTOMATED = 0
-    SEMI_AUTOMATED = 1
-    MANUAL = 2
-    SIMULATED = 3
-
-
-class YamlTestStep(YamlModelMixin, BaseModel):
-    label: str
-    PICS: Optional[str] = None
-    verification: Optional[str] = None
-    command: Optional[str]
-    disabled: bool = False
-    arguments: Optional[dict[str, Any]]
-
-
-class YamlTest(YamlModelMixin, BaseModel):
-    name: str
-    PICS: set[str] = set()
-    config: dict[str, Any]
-    steps: list[YamlTestStep] = Field(alias="tests")
-    type: YamlTestType = YamlTestType.MANUAL
-    path: Optional[Path]
+class YamlTest(YamlModelMixin, MatterTest):
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(steps=kwargs["tests"], **kwargs)
+        self.type: MatterTestType.MANUAL  # type: ignore
```

Comment:
Did you mean:

```suggestion
        self.type: MatterTestType = MatterTestType.MANUAL  # type: ignore
```
**or**
```suggestion
        self.type = MatterTestType.MANUAL  # type: ignore
```

Reply:
Thanks @antonio-amjr you are right.

Reply:
Also, there is another one like in the file:
`python_test_models.py`